Read archive path from Xcode preferences if set

taquitos opened the pull request:
- Detect when user has Xcode preferences and load the IDECustomDistributionArchivesLocation

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
Respects your Xcode preferences for Archive path

### Motivation and Context
Fixes https://github.com/fastlane/fastlane/issues/8729

Commit:
Issue 8729
- Detect when user has Xcode preferences and load the IDECustomDistributionArchivesLocation

KrauseFx commented on gym/lib/gym/detect_values.rb:
```diff
@@ -1,3 +1,4 @@
+require 'cfpropertylist'
```
Looks like that gem is actually not a direct dependency of fastlane, but of `xcodeprojc`, we should add it to `fastlane.gemspec` to be more explicit in case the dependency is dropped one day.

taquitos replied:
Good catch

KrauseFx commented on gym/lib/gym/detect_values.rb:
```diff
@@ -33,9 +34,34 @@ def self.set_additional_default_values
       config[:output_name] ||= Gym.project.app_name
 
+      if Gym.config[:build_path].nil? # maybe we manually passed it in somewhere?
+        config[:build_path] = archive_path_from_local_xcode_preferences
```
Or replace all those lines with
```ruby
config[:build_path] ||= archive_path_from_local_xcode_preferences
```

taquitos replied:
👍

KrauseFx commented on gym/lib/gym/detect_values.rb:
```diff
@@ -33,9 +34,34 @@ def self.set_additional_default_values
+      xcode_preferences_dictionary = CFPropertyList.native_types(CFPropertyList::List.new(file: path).value)
+      custom_archive_path = xcode_preferences_dictionary['IDECustomDistributionArchivesLocation']
+
+      unless custom_archive_path.to_s.length > 0
```
Could be
```ruby
return archive_path if custom_archive_path.to_s.length == 0
```

taquitos replied:
👍

KrauseFx commented on gym/lib/gym/options.rb:
```diff
@@ -228,6 +228,12 @@ def self.plain_options
                                      verify_block: proc do |value|
                                        UI.user_error!("File not found at path '#{File.expand_path(value)}'") unless File.exist?(value)
                                      end),
+        FastlaneCore::ConfigItem.new(key: :xcode_preference_plist_path,
```
Why do we need this option? We usually try to avoid adding options as much as possible, and only add ones that are helpful to a majority of our users 👍

taquitos replied:
We don't! I didn't realize you could do all this wonderful stubbing in Ruby so we don't need to inject anything via the options.
#rubyNoob

KrauseFx reviewed (changes requested):
This is great, I added some comments before it's ready 👍

Commit:
Updated to stub things during testing instead of adding options
Removed plist files that aren't needed thanks to stubbing
Added CFPropertyList dependency
Made the code more Ruby-ish

KrauseFx commented on fastlane.gemspec:
```diff
@@ -36,6 +36,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'terminal-notifier', '>= 1.6.2', '< 2.0.0' # macOS notifications
   spec.add_dependency 'terminal-table', '>= 1.4.5', '< 2.0.0' # Actions documentation
   spec.add_dependency 'plist', '>= 3.1.0', '< 4.0.0' # Needed for set_build_number_repository and get_info_plist_value actions
+  spec.add_dependency 'CFPropertyList', '~> 2.3.3' # Needed to be able to read binary plist format
```
I wouldn't be too strict about this, maybe have `'>= 2.3', '< 3.0.0'` instead. If `xcodeproj` bumps their dependency, it would result in unresolvable dependencies otherwise

taquitos replied:
👍

Commit:
Make CFPropertyList version requirement less strict

Commit:
Adds Joshua Liebowitz to authors

KrauseFx reviewed (approved):
Wohoo 🎷
